### PR TITLE
ARROW-14651: [Release][Archery] Add support for retrying download

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: dev/archery/archery/crossbow/cli.py [hunk 1]
```diff
@@ -339,7 +340,24 @@ def asset_callback(task_name, task, asset):
             path = target_dir / task_name / asset.name
             path.parent.mkdir(exist_ok=True)
             if not dry_run:
-                asset.download(path)
+                import github3
+                max_n_retries = 5
+                n_retries = 0
+                while True:
```
Comment (kszucs) [inline dev/archery/archery/crossbow/cli.py:346]:
```suggestion
                for n_retries in range(max_n_retries):
```

File: dev/archery/archery/crossbow/cli.py [hunk 2]
```diff
@@ -339,7 +340,24 @@ def asset_callback(task_name, task, asset):
...
+                    except github3.exceptions.GitHubException as error:
+                        n_retries += 1
+                        if n_retries == max_n_retries:
+                            raise
+                        wait_seconds = 60
+                        click.echo(f'Failed to download {path}')
+                        click.echo(f'Retry #{n_retries} after {wait_seconds}s')
+                        click.echo(error)
+                        time.sleep(wait_seconds)
+                        pass
```
Comment (kszucs) [inline dev/archery/archery/crossbow/cli.py:358]:
```suggestion
```

File: dev/archery/archery/crossbow/cli.py [hunk 3]
```diff
@@ -339,7 +340,24 @@ def asset_callback(task_name, task, asset):
...
-                asset.download(path)
+                import github3
+                max_n_retries = 5
+                n_retries = 0
+                while True:
+                    try:
+                        asset.download(path)
+                    except github3.exceptions.GitHubException as error:
+                        n_retries += 1
+                        if n_retries == max_n_retries:
```
Comment (kszucs) [inline dev/archery/archery/crossbow/cli.py:351]: Using a for loop you can remove this check.

Comment (kou) [reply dev/archery/archery/crossbow/cli.py:351]: If we use a for loop, no exception is raised when all retries are failed.

Comment (kszucs) [reply dev/archery/archery/crossbow/cli.py:351]: Right, didn't think of that :)

**Review by kszucs — APPROVED** [review]

File: dev/archery/archery/crossbow/cli.py (hunk 2 above)
Comment (kou) [reply dev/archery/archery/crossbow/cli.py:358]: Thanks!

**Comment by kou** [comment]
Ah, wait. There is a bug.

**Comment by kou** [comment]
> Ah, wait. There is a bug.

I misunderstood.

**Pull request closed without merging**